spinalcordtoolbox/deepseg/nnunet.py

```diff
@@ -15,13 +15,20 @@
 
 
 def create_nnunet_from_plans(path_model):
-    use_best_checkpoint = False
     tile_step_size = 0.5
     fold_dirs = [os.path.basename(path) for path in glob.glob(os.path.join(path_model, "fold_*"))]
     if not fold_dirs:
         raise FileNotFoundError(f"No 'fold_*' directories found in model path: {path_model}")
     folds_avail = 'all' if fold_dirs == ['fold_all'] else [int(f.split('_')[-1]) for f in fold_dirs]
 
+    # We prioritize 'checkpoint_final.pth', but fallback to 'checkpoint_best.pth' if not available
+    checkpoints = {os.path.basename(path) for path in glob.glob(os.path.join(path_model, "**", "checkpoint_*.pth"))}
+    for checkpoint_name in ['checkpoint_final.pth', 'checkpoint_best.pth', None]:
+        if checkpoint_name in checkpoints:
+            break
+    if checkpoint_name is None:
+        raise ValueError(f"Couldn't find 'checkpoint_final.pth' or 'checkpoint_best.pth' in {path_model}")
```

Comment:
FYI: It's maybe a little obscure, but this kind of searching loop, with code to run if no value was found, is exactly the use case for the [`for ... else`](https://docs.python.org/3/reference/compound_stmts.html#the-for-statement) loop in Python.

Reply:
Aha!! I had a feeling there was something fishy about the pattern I used there. I felt a little too clever about ending the loop on a `None`, so thank you for pointing that out. 

```suggestion
    for checkpoint_name in ['checkpoint_final.pth', 'checkpoint_best.pth']:
        if checkpoint_name in checkpoints:
            break  # Use the checkpoint that was found
    else:
        raise ValueError(f"Couldn't find 'checkpoint_final.pth' or 'checkpoint_best.pth' in {path_model}")
```

Or maybe just going for the simpler `if/else`, aha:

```suggestion
    if 'checkpoint_final.pth' in checkpoints:
        checkpoint_name = 'checkpoint_final.pth'
    elif 'checkpoint_best.pth' in checkpoints:
        checkpoint_name = 'checkpoint_best.pth'
    else:
        raise ValueError(f"Couldn't find 'checkpoint_final.pth' or 'checkpoint_best.pth' in {path_model}")
```

Reply:
Yep, that's exactly the change I had in mind :+1:

Reply:
Here's a third-party reference that goes further than the official docs and describes this exact pattern, too: 

- https://book.pythontips.com/en/latest/for_-_else.html#else-clause